### PR TITLE
material-ui: Use the react-router-dom switch

### PR DESCRIPTION
**Pull request by AlexITC**
This prevents that several components get rendered at the same time,
the switch will render the first matching component.

### CODE REVIEW

**Comment by oyvindberg** [comment]
Thanks! I'll test it locally within a few days, I'm curious if it solves all reloading issues both locally and when published. If you feel like it it would be nice to do the same change for the actual react router demo

**Comment by AlexITC** [comment]
There are other minor changes I was planning to submit but I forget, like the webpack config, and, adding a handler for unknown routes, which should solve the local issues. On the issues from the published app, you need that the server returns the index.html instead of a 404.

It can certainly be worth to add this to the react-router-demo

**Comment by AlexITC** [comment]
I have pushed the changes to fix the webpack config, and a global route handler to redirect to "/", that should fix the local reloading issues.

File: custom.webpack.config.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ var generated = require('./scalajs.webpack.config');
 
 var local = {
     module: {
+        devServer: {
```
Comment (oyvindberg) [inline custom.webpack.config.js:6]: seems you need to move this one level up

**Review by oyvindberg — APPROVED** [review]
I gave it a spin now. code changes look good, mui looks good (after I moved the `devServer` config), but reloading react-router-dom got me for instance `Cannot GET /topics` (probably like before).

Look into react-router-dom if you have the chance, otherwise this is good. I'll add you to the repo so you can merge it when you're happy

File: custom.webpack.config.js (hunk 1 above)
Comment (AlexITC) [reply custom.webpack.config.js:6]: Possibly, I'll complete the PR once I get some free time.

**Comment by AlexITC** [comment]
Merged manually.

**Pull request closed without merging**